### PR TITLE
split vm.Context into BlockContext and TxContext

### PR DESCRIPTION
**Pull request by gzliudan**
# Proposed changes

This PR splits vm.Context into BlockContext and TxContext.

Refreence: #21672

## Types of changes

What types of changes does your code introduce to XDC network?
_Put an `✅` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update (if none of the other choices apply)
- [ ] Regular KTLO or any of the maintaince work. e.g code style
- [ ] CICD Improvement

## Impacted Components
Which part of the codebase this PR will touch base on,

_Put an `✅` in the boxes that apply_

- [ ] Consensus
- [ ] Account
- [ ] Network
- [X] Geth
- [ ] Smart Contract
- [ ] External components
- [ ] Not sure (Please specify below)

## Checklist
_Put an `✅` in the boxes once you have confirmed below actions (or provide reasons on not doing so) that_

- [X] This PR has sufficient test coverage (unit/integration test) OR I have provided reason in the PR description for not having test coverage
- [ ] Provide an end-to-end test plan in the PR description on how to manually test it on the devnet/testnet.
- [ ] Tested the backwards compatibility.
- [ ] Tested with XDC nodes running this version co-exist with those running the previous version.
- [ ] Relevant documentation has been updated as part of this PR
- [ ] N/A

### CODE REVIEW

File: core/vm/evm.go [hunk 1]
```diff
@@ -221,7 +234,7 @@ func (evm *EVM) Call(caller ContractRef, addr common.Address, input []byte, gas
 		return nil, gas, ErrDepth
 	}
 	// Fail if we're trying to transfer more than the available balance
-	if !evm.Context.CanTransfer(evm.StateDB, caller.Address(), value) {
+	if value.Sign() != 0 && !evm.Context.CanTransfer(evm.StateDB, caller.Address(), value) {
```
Comment (wgr523) [inline core/vm/evm.go:237]: is `value.Sign() != 0` necessary? I don't see this in ethereum https://github.com/ethereum/go-ethereum/pull/21672

File: eth/tracers/tracer.go [hunk 2]
```diff
@@ -670,7 +670,7 @@ func (jst *JsTracer) CaptureStart(env *vm.EVM, from common.Address, to common.Ad
 	jst.ctx["to"] = to
 	jst.ctx["input"] = input
 	jst.ctx["gas"] = gas
-	jst.ctx["gasPrice"] = env.Context.GasPrice
+	jst.ctx["gasPrice"] = env.GasPrice
```
Comment (wgr523) [inline eth/tracers/tracer.go:673]: I think tracer doesn't need gasPrice. Do we need it?

File: core/vm/evm.go (hunk 1 above)
Comment (gzliudan) [reply core/vm/evm.go:237]: It's in #21222:

[```](https://github.com/holiman/go-ethereum/blob/8528951f5477600ce703bdc37b1e3071b5619b8c/core/vm/evm.go#L204-L206)

File: eth/tracers/tracer.go (hunk 2 above)
Comment (gzliudan) [reply eth/tracers/tracer.go]: I think tracer need this field. This line was submitted #558 and also exist in geth.

**Review by wgr523 — APPROVED** [review]

**Pull request merged**